packages/steps/src/utils/template.ts

```diff
@@ -19,6 +24,42 @@ export function interpolateWithOutputs(
   return interpolate(templateString, BUILD_STEP_OUTPUT_EXPRESSION_REGEXP, fn);
 }
 
+export function getObjectValueForInterpolation(
+  pathWithObjectName: string,
+  obj: Record<string, any>
+): string | number | boolean | null {
+  const value = get(obj, pathWithObjectName.split('.').slice(1).join('.'));
```

Comment:
maybe to avoid hacks like that provide object that already has a top level field needed there?